Cache/Main.lean

```diff
@@ -45,11 +45,9 @@ def main (args : List String) : IO Unit := do
   | ["put!"] => putFiles (← mkCache hashMap false) true (← getToken)
   | ["commit"] =>
     if !(← isStatusClean) then IO.println "Please commit your changes first" return else
-    putFiles (← mkCache hashMap false) false (← getToken)
```

Comment:
I've made `put` and `commit` independent commands so that we can do `put` even if `commit` would fail.  This can happen when people forget to add files to `Mathlib.lean` (which is then modified at the beginning of the build).

We also do `commit` *before* `put` now.  This should help with garbage collection races.